Enable Service Affinity for OpenStack cloudprovider

FengyunPan opened the pull request:
Fix issue: #48500
Kubernetes's OpenStack cloudprovider can't set persistence to "SOURCE_IP"

**Release note**:
```release-note
NONE
```

dims commented on pkg/cloudprovider/providers/openstack/openstack_loadbalancer.go:
```diff
@@ -546,15 +546,6 @@ func (lbaas *LbaasV2) createLoadBalancer(service *v1.Service, name string) (*loa
 	return loadbalancer, nil
 }
 
-func stringInArray(x string, list []string) bool {
```
This seems like an unrelated change. Can you please file as a separate PR?

FengyunPan replied:
Ok, done.

dims commented:
/ok-to-test

Commit:
Enable Service Affinity for OpenStack cloudprovider.

Fix issue: #48500
Kubernetes's OpenStack cloudprovider can't set LB's persistence
to "SOURCE_IP".

anguslees commented:
Oops ;)
/lgtm

dims commented:
/approve
/lgtm

dims commented:
/test pull-kubernetes-e2e-gce-etcd3
